routes/awsapi.js

```diff
@@ -0,0 +1,132 @@
+/*
+This file holds all aws logic
+It handles file upload signing and SNS notification
+*/
+
+const aws = require('aws-sdk');
+const request = require('request');
+const express = require('express');
+const uuidv1 = require('uuid/v1');
+var router = express.Router();
+
+
+/*
+  set envirnoment variables: ACCESS_KEY_ID, SECRET_ACCESS_KEY
+*/
+
+const S3_BUCKET         = "upload.media.futureboard.olin.build";
+const ACCESS_KEY_ID     = process.env.ACCESS_KEY_ID;
+const SECRET_ACCESS_KEY = process.env.SECRET_ACCESS_KEY;
+
+if(ACCESS_KEY_ID == null || SECRET_ACCESS_KEY == null){
+  console.log("ERROR: s3 envirnoment variables not set!");
+}
+
+aws.config.update({
+  accessKeyId: ACCESS_KEY_ID,
+  secretAccessKey: SECRET_ACCESS_KEY
+});
+
+
+module.exports = function (io, db) {
+  router.route('/s3-sign')
+    .get(GETSignRequest)
+    .post(POSTS3Sign);
+
+  router.route('/MediaStatusSNS')
+    .post(POSTSNSNotify);
+
+  /*
+   * Respond to GET requests to /sign-s3.
+   * Upon request, return JSON containing the temporarily-signed S3 request and
+   * the anticipated URL of the image.
+   */
+  function GETSignRequest(req, res) {
+    const s3 = new aws.S3({
+        signatureVersion: 'v4',
+        // Endpoint can be switched to https://media.futureboard.olin.build/' after we get an ssl cert for media.FUTUREboard.olin.build
+        endpoint: new aws.Endpoint('https://s3.amazonaws.com')
+      });
+    const inputFileName = req.query['file-name'].split('.');
+    const fileType = req.query['file-type'];
+    const mediaType = fileType.split('/')[0]; // Get image or video
+    const inputFileExtension = inputFileName[inputFileName.length - 1];
+    let uploadFileName = '';
+    let outputFileName = '';
+    const new_uuid = uuidv1();
+
+    if(mediaType === 'image'){
```

Comment:
can we do `if (...) {` (with spaces) as the standard syntax? Is there any reason for no spaces?